packages/lib/services/share/ShareService.ts

```diff
@@ -118,11 +119,15 @@ export default class ShareService {
 				throw new Error('The local public private key pair uses an unsupported algorithm. It may be necessary to upgrade Joplin or share from a different device.');
 			}
 
-			// TODO: handle "undefinedMasterPassword" error - show master password dialog
-			folderMasterKey = await this.encryptionService_.generateMasterKey(getMasterPassword());
-			folderMasterKey = await MasterKey.save(folderMasterKey);
+			if (folder.master_key_id) {
+				logger.debug(`Folder ${folderId}'s already has a master key. Not creating a new one.`);
```

Comment:
That's probably ok to make this an `info` statement. I think `debug` we can reserve for things that will happen too frequently and fill up the log, such as sync events